src/java.base/share/classes/java/lang/foreign/Arena.java

```diff
@@ -265,10 +265,10 @@ static Arena ofShared() {
      * aligned according the provided alignment constraint.
      *
      * @implSpec
-     * Implementations of this method must return a native segment featuring the
-     * requested size, and that is compatible with the provided alignment constraint.
-     * Furthermore, for any two segments {@code S1, S2} returned by this method, the
-     * following invariant must hold:
+     * Implementations of this method must return a native, zero-initialized segment
+     * featuring the requested size, and that is compatible with the provided alignment
```

Comment:
Do we want to _mandate_ that all arenas return zero-initialized memory? Maybe it's enough to say that the default implementation does this.

Reply:
It's not a default method. That said, according to this suggestion it could be possible to move the text in the ofAuto/ofConfined/ofArena factories.

Reply:
The choice here is: if we allow non-zeroing implementation, implementors of Arena have more flexibility, but clients have less guarantees (how does a client know if what they get back is zeroed?)

Reply:
> (how does a client know if what they get back is zeroed?)

It seems similar to e.g. a HashMap vs. LinkedHashMap with regards to ordering. The creator of the Arena would decide the zeroing strategy.

Reply:
Not sure if that analogy is 100% correct. In that case a client that takes a `Map`, but wants order, can say to accept e.g. a `SequencedMap`. In the case of `Arena`, there's no subtype which adds the "zeroing" characteristics. So a client accepting an arena, and doing some allocation would be effectively at the mercy of who created the arena. I think one can argue this both ways.